frontend/src/hooks/useDashboardData.js

```diff
@@ -0,0 +1,112 @@
+import { useQuery } from '@tanstack/react-query';
+import { useState, useEffect } from 'react';
+import { axiosInstance } from 'utils/axios';
+import { ZETH_ADDRESS } from 'utils/constants';
+import { ReactComponent as CollateralIcon } from 'assets/icons/collateral.svg';
+import { ReactComponent as EthIcon } from 'assets/icons/ethereum.svg';
+import { ReactComponent as UsdIcon } from 'assets/icons/usd_coin.svg';
+import { ReactComponent as BorrowIcon } from 'assets/icons/borrow.svg';
+import { ReactComponent as StrkIcon } from 'assets/icons/strk.svg';
+
+const QueryKeys = {
+  DashboardData: 'dashboardData',
+};
+
+const fetchCardData = async (walletId) => {
+  if (!walletId) {
+    console.error('fetchCardData: walletId is undefined');
+    return null;
+  }
+  try {
+    const response = await axiosInstance.get(`/api/dashboard?wallet_id=${walletId}`);
+    return response.data;
+  } catch (error) {
+    console.error('Error during getting the data from API', error.response ? error.response.data : error.message);
+    return null;
+  }
+};
+
+const useDashboardData = (walletId) => {
+  const [healthFactor, setHealthFactor] = useState('0.00');
+  const [currentSum, setCurrentSum] = useState(0);
+  const [cardData, setCardData] = useState([
+    {
+      title: 'Collateral & Earnings',
+      icon: CollateralIcon,
+      balance: '0.00',
+      currencyName: 'Ethereum',
+      currencyIcon: EthIcon,
+    },
+    {
+      title: 'Borrow',
+      icon: BorrowIcon,
+      balance: '0.00',
+      currencyName: 'USD Coin',
+      currencyIcon: UsdIcon,
+    },
+  ]);
+  const [loading, setLoading] = useState(true);
```

Comment:
why do we need this loading state here? I believe data is already loaded from a useQuery hook and accessible in query.data